src/blenderbim/blenderbim/bim/module/model/prop.py

```diff
@@ -87,6 +87,41 @@ def update_type_page(self, context):
     AuthoringData.data["paginated_relating_types"] = AuthoringData.paginated_relating_types()
 
 
+def update_relating_array_from_object(self, context):
+    if self.relating_array_object is None:
+        return
+    element = tool.Ifc.get_entity(self.relating_array_object)
+    if not element:
+        return
+    # If a child is selected, the following assigns the parent to the relating_array_object
+    parent_globalid = element.IsDefinedBy[0].RelatingPropertyDefinition.HasProperties[0].NominalValue.wrappedValue
+    parent_element = tool.Ifc.get().by_guid(parent_globalid)
+    parent_of_relating_array_object = tool.Ifc.get_object(parent_element)
+    #The following error handling is used to break out when an array parent is selected
+    try:
+        element.IsDefinedBy[0].RelatingPropertyDefinition.HasProperties[1].NominalValue.wrappedValue
+    except AttributeError:
+        pass
+    else:
+        bpy.ops.bim.enable_editing_array(item=self.is_editing)
+        return 
+
+    self.relating_array_object = parent_of_relating_array_object
+    bpy.ops.bim.enable_editing_array(item=self.is_editing)
+
+    return self.relating_array_object
```

Comment:
I was getting an error that the function should return `None`. 
```suggestion
    return
```